nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -173,8 +185,18 @@ def _apply(
         dataset: Optional[Dataset] = None,
     ) -> TModel:
         modified_model = copy_model(model)
+        backend = get_backend(modified_model)
+
         if statistic_points is None:
-            backend = get_backend(modified_model)
+            for algorithm in self.first_stage_algorithms:
+                if isinstance(algorithm, SmoothQuantize) and backend != BackendType.OPENVINO:
```

Comment:
It looks like, this is redundant check because SmoothQuantize algorithm filed with the same error.

Reply:
The algorithm raises `RuntimeError` for now. We don't need such behavior for `ONNX` models with `model_type: transformer`, I think.